http.go

```diff
@@ -12,47 +13,88 @@ import (
 )
 
 type S2SClientConfig struct {
-	Service       string
+	ServiceName   string
 	JWTSecret     string
 	DebugRequests bool
+	Expiration    time.Duration
+}
+
+func (cfg *S2SClientConfig) Validate() error {
+	if cfg.JWTSecret == "" {
+		return ErrEmptyJWTSecret
+	}
+
+	return nil
 }
 
 // Service-to-service HTTP client for internal communication between Sequence services.
-func S2SClient(cfg *S2SClientConfig) *http.Client {
+func S2SClient(cfg *S2SClientConfig) (*http.Client, error) {
+	if cfg == nil {
+		return nil, ErrS2SClientConfigIsNil
+	}
+
+	if cfg.JWTSecret == "" {
+		return nil, ErrEmptyJWTSecret
+	}
+
+	tokenCfg := &S2STokenConfig{
+		JWTSecret:   cfg.JWTSecret,
+		ServiceName: cfg.ServiceName,
+		Expiration:  cfg.Expiration,
+	}
+
 	httpClient := &http.Client{
 		Transport: transport.Chain(http.DefaultTransport,
 			traceid.Transport,
-			transport.SetHeaderFunc("Authorization", s2sAuthHeader(cfg.JWTSecret, map[string]any{"service": cfg.Service})),
+			transport.SetHeaderFunc("Authorization", s2sAuthHeader(tokenCfg)),
 			transport.If(cfg.DebugRequests, transport.LogRequests(transport.LogOptions{Concise: true, CURL: true})),
 		),
 	}
 
-	return httpClient
+	return httpClient, nil
 }
 
-// Create short-lived service-to-service JWT token for internal communication between Sequence services.
-func S2SToken(jwtSecret string, claims map[string]any) string {
-	jwtAuth := jwtauth.New("HS256", []byte(jwtSecret), nil, jwt.WithAcceptableSkew(2*time.Minute))
+func s2sAuthHeader(cfg *S2STokenConfig) func(req *http.Request) string {
+	return func(req *http.Request) string {
+		return "BEARER " + S2SToken(cfg)
+	}
+}
 
-	now := time.Now().UTC()
+const (
+	defaultExpiration time.Duration = 30 * time.Second
+	acceptableSkew    time.Duration = 2 * time.Minute
+)
+
+type S2STokenConfig struct {
+	JWTSecret   string
+	ServiceName string
+	Expiration  time.Duration
+}
 
-	c := maps.Clone(claims)
-	if c == nil {
-		c = map[string]any{}
+func (cfg *S2STokenConfig) Validate() error {
+	if cfg.JWTSecret == "" {
+		return ErrEmptyJWTSecret
 	}
 
-	c["iat"] = now
+	return nil
+}
 
-	if _, ok := c["exp"]; !ok {
-		c["exp"] = now.Add(30 * time.Second)
+// Create short-lived service-to-service JWT token for internal communication between Sequence services with HS256 algorithm.
+func S2SToken(cfg *S2STokenConfig) string {
+	if cfg == nil {
+		return ""
 	}
```

Comment:
`return ""` doesn't feel right

how about we make the cfg a non-pointer? and call it TokenClaims?

Reply:
I suggest `(string, bool)`

Reply:
It panics, when it is nil
When the JWTSecret is empty, then it will return `""` in the end.

I added validate function, which checks JWTSecret.
Best would be, if this function can return error.